### ISSUE TITLE
Introduce a flag to retry test failures only if TEST_INFRASTRUCTURE_FAILURE_FILE is written

### ISSUE THREAD

**Issue by tomrenn** [bazelbuild/bazel#13588]
### Description of the problem / feature request:

Google's internal version of bazel, has a flag called --treat_test_infrastructure_failures_as_flaky, which retries tests up to 3 times (similar to Flaky=True). This only happens if the test output contains a file in the $TEST_INFRASTRUCTURE_FAILURE_FILE as defined in the [bazel docs](https://docs.bazel.build/versions/main/test-encyclopedia.html#initial-conditions). 

### Feature requests: what underlying problem are you trying to solve with this feature?

The feature is trying to improve tests which may have flaky infrastructure, but are generally reliable otherwise. For example, we have some tests which relying on starting an emulator for testing. If this emulator fails to start, we'd like to treat that as an infrastructure failure and indicate the test should be retried.

This is an improvement over using Flaky=True on the test target, because there can be other test failures we would prefer not to be retried. `--treat_test_infrastructure_failures_as_flaky` would give tests the ability to have flaky retries only during certain [infrastructure] failures.

**Comment by joeljeske** [comment]
This still seems valuable, and a great addition to the bazel test story. @bazelbuild/triage

**Comment by EdSchouten** [comment]
@tomrenn @meisterT Is adding support for this merely a matter of taking the internal implementation of `--treat_test_infrastructure_failures_as_flaky` and adding it to the public version of Bazel, or is there more to it? In case it's the former, would Google be interested in dropping the code for that?

**Comment by meisterT** [comment]
cc @lberki @haxorz 

@tomrenn do you want to take this one yourself?

**Comment by EdSchouten** [comment]
Friendly ping!

**Comment by tomrenn** [comment]
Yes, this would be taking the internal implementation and making it available to Bazel as well. I think it should be a straightforward change but I won't have time to do this before 7.0.0 (in October). I can try to do this before the end of the year but no promises.

**Comment by acecilia** [comment]
@tomrenn Any updates about this? 🙏  (I guess only Google employees could do this, as the internal implementation is not public?) 

The other option would be to implement it from scratch

**Comment by meteorcloudy** [comment]
Internally, this flag isn't implemented in the build system but in some remote build service. I think it makes sense to implement from scratch in Bazel. @acecilia Are you willing to contribute a PR?

**Comment by acecilia** [comment]
Understood @meteorcloudy thanks so much for that piece of context. Yes I am willing to contribute, will do so if I get to it. Thanks again 🙏

**Comment by acecilia** [comment]
Not stale
